src/Corax/Querying/IndexSearcher.cs

```diff
@@ -120,18 +120,18 @@ private IndexSearcher(IndexFieldsMapping fieldsMapping)
             _fieldMapping = fieldsMapping;
         }
     }
-    
-    public EntryTermsReader GetEntryTermsReader(long id, ref Page p)
+
+    public void GetEntryTermsReader(long id, ref Page p, out EntryTermsReader reader, CompactKey existingKey = null)
     {
         if (_entryIdToLocation.TryGetValue(id, out var loc) == false)
             throw new InvalidOperationException("Unable to find entry id: " + id);
 
         InitializeSpecialTermsMarkers();
         
         var item = Container.MaybeGetFromSamePage(_transaction.LowLevelTransaction, ref p, loc);
-        return new EntryTermsReader(_transaction.LowLevelTransaction, _nullTermsMarkers, _nonExistingTermsMarkers, item.Address, item.Length, _dictionaryId);
+        reader = new EntryTermsReader(_transaction.LowLevelTransaction, _nullTermsMarkers, _nonExistingTermsMarkers, item.Address, item.Length, _dictionaryId, existingKey);
```

Comment:
`EntryTermsReader` is disposable now as I see. Who is going to dispose this instance?

Reply:
I think we're missing  the disposal of this by the callers

Reply:
Most cases we don't need it. This is an optimization. I did however take care of the case where we don't own the key, which could introduce subtle defects if used improperly.

Reply:
If the disposal isn't needed in _most_ cases then does it make sense to make it `IDisposable`? Shouldn't we handle then the case you mentioned externally? Especially that it "could introduce subtle defects", so maybe it makes more sense to handle this case explicitly instead of delegating it to the disposal of the reader?

Reply:
I see that I haven't pushed the changes. I am now handling it, but I think we can do better than this. I will prototype another solution and see if I can make it more consistent instead.